core/sail/memory/src/main/java/org/eclipse/rdf4j/sail/memory/MemoryStore.java

```diff
@@ -366,6 +373,25 @@ public ValueFactory getValueFactory() {
 		return store.getValueFactory();
 	}
 
+	@Override
+	public Map<IRI, Literal> getSettings() {
+		HashMap<IRI, Literal> settings = new HashMap<>();
+		settings.put(MemoryStoreSchema.SYNC_DELAY, SimpleValueFactory.getInstance().createLiteral(syncDelay));
+		return settings;
+	}
+
+	@Override
+	public void setSettings(Map<IRI, Literal> settings) {
+		settings.forEach((key, value) -> {
+
+			if (key.equals(MemoryStoreSchema.SYNC_DELAY)) {
+				setSyncDelay(value.longValue());
+			}
+
+		});
+
+	}
```

Comment:
Here is part of my solution proposal. The repository layer uses an immutable RepositorySettings object, but the sail layer just uses a Map<IRI, Literal>. The map is created by the lowest level sail, and each wrapping sail can add to the map. We reuse the vocabulary from the sail config.